### DISCUSSION TITLE
DBeaver fetch size limitation

### DISCUSSION THREAD

**Discussion by Beekerc69** [dbeaver/dbeaver#23283] [General]
i'm trying to query the information_schema.columns table of my environment and it has about 37,000 rows.  
no matter what i set fetch size to (i tried huge numbers, and i tried zero), on the toolbar or the Preferences Data Editor, it will only return 1,000 rows.  
how do i get around this limitation?

**Comment by LonwoLonwo** [comment]
Hello @Beekerc69 

You can set this (result set fetch) value to -1 as an option

![2024-04-02 10_37_03-Preferences](https://github.com/dbeaver/dbeaver/assets/45152336/8ec676a9-ece6-4cfe-acc3-3f18d5b25fa5)

**Reply by Beekerc69** [reply]
Thanks LonwoLonwo.
but that didn't work.  
i put -1 in the Parameter page as indicated.  also tried with "Use SQL to limit fetch size" both checked and unchecked.
when i put -1 in the fetch size in the toolbar, it returns 500 rows.
when i put 0 in the fetch size in the tool bar, it returns 500 rows.
when i put 50000 in the fetch size in the tool bar, it returns 1000 rows.
but that's the most i can get it to return.

**Reply by LonwoLonwo** [reply]
I can't reproduce, unfortunately. I set the fetch size parameter to 5000, and DBeaver reads 5000 rows now. Are you sure that it is not a database limitation? Also, you can check your driver properties in your connection settings.

**Reply by Beekerc69** [reply]
back end is postgresql and using a jdbc:jaspersoft URL connector.  how would i find out if it's a database limitation?
what should i be looking for in driver properties?  
my background is using SQLDeveloper connecting to Oracle databases with TNSNAMES info and SIDs.  
i understand the SQL stuff.  it's this using dBeaver and how configure drivers/connectors that's a bit alien to me.
if you can impart this troubleshooting wisdom through the "for dummies" filter, that would really be helpful.  
thanks
